Bloom.md

```diff
@@ -0,0 +1,341 @@
+are possible, but false negatives are not possible. https://en.wikipedia.org/wiki/Bloom_filter
+
+To utilize Bloom filters in their client applications, users today use client libraries that are compatible with the ReBloom
+including jedis, redis-py, node-redis, nredisstack, rueidis, rustis, etc. This allows customers to perform bloom filter
+based operations, e.g. add and set.
+
+Redis Ltd.‘s ReBloom is published under a proprietery license and hence cannot be distributed freely with ValKey.
+
+There is growing [demand](https://github.com/orgs/valkey-io/discussions?discussions_q=+bloom+) for an
+(1) Open Source bloom filter feature in Valkey which is (2) compatible with the ReBloom API syntax and with
+existing ReBloom based client libraries. ValkeyBloom will help address both these requirements.
+
+
+## Design Considerations
+
+The ValkeyBloom module brings in a bloom module data type into Valkey and provides commands to create / reserve
+bloom filters, operate on them (add items, check if items exist), inspect bloom filters, etc.
+It allows customization of properties of bloom filter (capacity, false positive rate, expansion rate, specification of
+scaling vs non-scaling filters etc) through commands and configurations. It also allows users to create scalable bloom
+filters and back up & restore bloom filters (through RDB load and save).
+
+ValkeyBloom provides commands (BF.*), configs, etc to operate on Bloom objects which are top-level structures containing
+lower-level BloomFilter provided by an external crate. ValkeyBloom utilizes an open source (BSD-2) Bloom Filter Rust library
+around which it implements a scalable bloom filter.
+
+When a bloom filter is created, a bit array is created with a length proportional to the capacity (number of items the user
+wants to add to the filter) and hash functions are also created. The number of hash functions are controlled by the
+false positive rate that the user configures.
+
+When a user adds an item (e.g. BF.ADD) to the filter, the item is passed through the filter and corresponding bit are set
+to 1. When a user checks whether an item exists on a filter (e.g. BF.EXISTS), the item is passed through the filters
+and if all the resolved bits have values as 1, we can say that the item exists with a false positive rate of
+X (specified by the user when creating the filter). If any of the bits are 0, the item does not exist and the BF.EXISTS
+operation will return 0.
+
+We have the following terminologies / properties:
+* Bloom Object: The top level structure representing the data type. It contains meta data and a list of lower-level
+                Bloom Filters (Implemented by an external crate) in case of scaling and a single lower-level bloom filter
+                in case of non scaling.
+* Bloom Filter: A single bloom filter (Implemented by an external crate).
+* Capacity: The number of items we expect to add to a bloom filter. This controls the size of the filter.
```

Comment:
number of unique items? 
I think you meant to say, number of unique items that we want to be able to detect without false positives?

Reply:
This refers to the number of items the user wants to add to a bloom object - the add operations itself is based on the configured false positive rate - because add operations can be rejected if we detect (check is based on false positive rate) that the item exists already.